src/model/RegexStructure.ts

```diff
@@ -0,0 +1,16 @@
+/**
+ * The same Enum as defined in the backend.
+ * Please consult the backend for documentation of the fields.
+ */
+enum RegexStructure {
+  CHARACTER_CLASS,
+  CHARACTER_CLASS_RANGE,
+  NEGATED_CHARACTER_CLASS,
+  SINGLE_CHARACTER_MATCH,
+
+}
+
+function getRegexStructureFromString(string: string): RegexStructure {
+  if (!Object.values(RegexStructure).includes(string)) throw "this regex structure doesnt exist"
+  return Object.entries(RegexStructure).find(entry => entry[0] === string)![1] as RegexStructure
```

Comment:
… I have absolutely no idea what exactly you are doing here.

Reply:
Exactly what the method name says: Converting a string to the regex structure equivalent.
So "`CHARACTER_CLASS`" will map to `0`.